### PR TITLE
⬆️ Update mint module to align on latest cosmos-sdk v0.50.6

### PR DESCRIPTION
**Pull request by bdeneux**
This PR includes the necessary updates to the Mint module to align it with the Cosmos-SDK v0.50.6. :gear:  

Key changes include:  
- Updated the Keeper struct and its methods to match the new Cosmos-SDK version. :wrench:
- Kept our custom mint implementation intact while ensuring compatibility with the new SDK version. :shield:
- Updated the necessary dependencies in go.mod. :package:

## Summary by CodeRabbit

- **Refactor**
  - Improved code consistency and readability by renaming variables and refining method signatures across the mint module.
  - Updated import paths and reorganized code structure for better maintainability.

- **New Features**
  - Introduced `NewAppModule` function with updated parameters and logic.
  - Added methods to implement `depinject.OnePerModuleType` and `appmodule.AppModule` interfaces.

- **Bug Fixes**
  - Enhanced validation logic in `ValidateGenesis` and `ValidateMinter` functions to ensure more accurate genesis state validation.

- **Tests**
  - Updated test functions to align with new method signatures and parameter changes for consistent testing practices.

### CODE REVIEW

**Review by amimart — APPROVED** [review]
LGTM! Thanks 🙏

**Review by ccamel — APPROVED** [review]
Great! Thanks :relieved:

File: x/mint/types/minter.go [hunk 1]
```diff
@@ -48,11 +48,11 @@ func DefaultInitialMinter() Minter {
...
 
-// Validate validates the mint parameters.
-func (m Minter) Validate() error {
-	if m.Inflation.IsNegative() {
+// ValidateMinter does a basic validation on minter.
+func ValidateMinter(minter Minter) error {
+	if minter.Inflation.IsNegative() {
 		return fmt.Errorf("mint parameter Inflation should be positive, is %s",
-			m.Inflation.String())
+			minter.Inflation.String())
```
[the remark this answers is not in the document]
Comment (bdeneux) [reply x/mint/types/minter.go:55]: @coderabbitai This function is not directly tested but there are indirect test in `minter_test.go` that check negative inflation.

File: x/mint/types/genesis.go [hunk 2]
```diff
@@ -23,5 +23,5 @@ func ValidateGenesis(data GenesisState) error {
 		return err
 	}
 
-	return data.Minter.Validate()
+	return ValidateMinter(data.Minter)
```
[the remark this answers is not in the document]
Comment (bdeneux) [reply x/mint/types/genesis.go:26]: https://github.com/coderabbitai This function is not directly tested but there are indirect test in minter_test.go that already check negative inflation.

**Pull request merged**